assets/src/utils/minio_utils.py

```diff
@@ -6,30 +6,60 @@
 import pdf2image.exceptions
 import PIL.Image
 import urllib3.exceptions
-from minio.credentials import AWSConfigProvider
+from minio.credentials import AWSConfigProvider, EnvAWSProvider, IamAwsProvider
 
 from src import db, logger
 from src.config import settings
 
 logger_ = logger.get_logger(__name__)
 
-minio_config = {
-    "endpoint": settings.s3_endpoint,
-    "secure": settings.minio_secure_connection,
-}
-if settings.aws_profile_name is not None:
-    # 'minio' library is used to connect with AWS S3
-    minio_config.update(
-        {"credentials": AWSConfigProvider(profile=settings.aws_profile_name)}
+
+class NotConfiguredException(Exception):
+    pass
+
+
+minio_config = {}
+
+minio_config.update({"secure": settings.minio_secure_connection})
+
+if settings.s3_endpoint:
+    minio_config.update({"endpoint": settings.s3_endpoint})
+
+if settings.s3_credentials_provider == "minio":
+    logger_.info(
+        f"S3_Credentials provider - " f"{settings.s3_credentials_provider}"
     )
-else:
-    # 'minio' library is used to connect with Minio service locally
     minio_config.update(
         {
             "access_key": settings.s3_access_key,
             "secret_key": settings.s3_secret_key,
         }
     )
+elif settings.s3_credentials_provider == "aws_iam":
+    logger_.info(
+        f"S3_Credentials provider - {settings.s3_credentials_provider}"
+    )
+    minio_config.update({"credentials": IamAwsProvider()})
+elif settings.s3_credentials_provider == "aws_env":
+    logger_.info(
+        f"S3_Credentials provider - {settings.s3_credentials_provider}"
+    )
+    minio_config.update({"credentials": EnvAWSProvider()})
+elif settings.s3_credentials_provider == "aws_config":
+    # environmental variable AWS_PROFILE_NAME should be set
+    logger_.info(
+        f"S3_Credentials provider - {settings.s3_credentials_provider}"
+    )
+    minio_config.update(
+        {"credentials": AWSConfigProvider(profile=settings.aws_profile_name)}
+    )
+else:
+    raise NotConfiguredException(
+        "s3 connection is not properly configured - "
+        "s3_credentials_provider is not set"
+    )
```

Comment:
Let's make it like you've done in annotations.

Reply:
agreed